src/runtime.rs

```diff
@@ -41,6 +43,13 @@ impl std::fmt::Debug for Revision {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct RevisionControlSystem {
```

Comment:
nit: RevisionController

Can you move this type to another file, `src/runtime/controller.rs`?